### ISSUE TITLE
[docs] android activity recognition permissions

### ISSUE THREAD

**Issue by maki3000** [expo/expo#13271]
### Summary

Hello! I'm trying to build an app that uses the google fit API and I have troubles to set the permissions. I added following permissions to our app.json:

`"permissions": 
        ["com.google.android.gms.permission.ACTIVITY_RECOGNITION", "android.permission.ACTIVITY_RECOGNITION", "ACTIVITY_RECOGNITION", "VIBRATE"]`

None of it is working. We did make a google fit API key in the cloud console of google, but I don't know how to connect this key with our expo build. Or is it enough, if you put the SHA1 fingerprint from expo in there?

Anyway, with old android system, our pedometer works fine, but with the newer systems (Android 10 and above), it doesn't even ask for accessing the fit data?

I ask for accessing the pedometer as it is in the documentation, like this:

`const { status } = await Pedometer.requestPermissionsAsync();`
And then check if status is granted and it seems it is, but pedometer delivers no data.

There is this line on bottom of pedometer documentation:
https://docs.expo.io/versions/latest/sdk/pedometer/#standalone-applications
And we tried to follow the instructions, but the documentation on how to connect expo build with google fit API is unclear to me.

### What platform(s) does this occur on?
Android

### SDK Version (managed workflow only)
41.0.1

### Testing device
Nokia 7.2 / Android 10

### Link to the related docs page

https://docs.expo.io/versions/latest/config/app/#permissions

**Comment by FabiGu** [comment]
Same problem here...

**Comment by maki3000** [comment]
UPDATE:

We cleaned our Android permissions to just this:
```["com.google.android.gms.permission.ACTIVITY_RECOGNITION", "VIBRATE"]```

And we filled out all the guidelines/App-contents (privacy policy and stuff).

Now it seems to work better with permissions.

There is still a problem remaining with following code:
```
const checkSettingsAsync = async () => {
    const { status } = await Pedometer.requestPermissionsAsync();
    if (status !== 'granted') {
      Toast.show({
        text: "Sorry! Your sensors aren't activated. Please check that you've allowed them for the Synchronisator.",
        buttonText: "Okay",
        position: "top",
        type: "danger",
        duration: 10000,
      });
    } else {
      navigation.navigate("Walk");
    }
  }
```
The `status` seems always to be `granted` on Android, even if permissions are denied in the settings/permissions-manager. On iOS this works fine.

We'd be very glad for help here!

**Comment by ArchieGregory** [comment]
Was this ever fixed?
Having a similar issue on Android 12

**Comment by maki3000** [comment]
Here's is what we did so far to make this work on Android:

- Make a YouTube video, where you explain, what you do with the collected google fitness data
- Put the privacy policy on a responsive website
- Put the Android permissions as mentioned above

Then our code now looks like this:
```
const checkSettingsAsync = async () => {
    const { granted } = await Pedometer.getPermissionsAsync();
    if (!granted) {
        // show an alert or something
    } else {
        // do something with Pedometer
    }
};
```
We call our `checkSettingsAsync` on a button press, where we either let the user to another screen or show a warning.

We just told google, that we did all this and hope that we now get our permissions...

**Comment by maki3000** [comment]
We finally did the update, but with no luck. No permissions are showing up and the `const { granted } = await Pedometer.getPermissionsAsync();` thinks it's granted on Android version more than 10.

**Comment by maki3000** [comment]
Finally we got it to work with the Android ACTIVITY_RECOGNITION permission.

Here is what we did:

- Inside app.config.json we added the permissions like following:

`
"permissions": [
                "com.google.android.gms.permission.ACTIVITY_RECOGNITION",
                "android.permission.ACTIVITY_RECOGNITION"
]
`

I really don't know which of these permissions is needed. In the expo.dev documentation they say the first one, in Android documentation, they say the latter.

What we also did, is make use of the expo config plugins. Here's the example:
https://docs.expo.dev/guides/config-plugins/#modifying-the-androidmanifestxml
There we added the permission `android.permission.ACTIVITY_RECOGNITION`.

Unfortunately we don't know, what exactly now made it work. If I find that out, I will post it here of course.

BTW: This one, mentioned in a earlier post, is not needed at all ->
- Make a YouTube video, where you explain, what you do with the collected google fitness data
This would be, if you use Googles fitness data for a web app.
